rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/queue/RedisCircularBuffer.java

```diff
@@ -0,0 +1,220 @@
+
+    @Override
+    public boolean addAll(Collection<? extends ValueType> toAdd) {
+        try (Handle handle = rdbi.open()) {
+            for (final ValueType value : toAdd) {
+                final String valueAsString = serializationHelper.encode(value);
+                final boolean success = handle.attach(RedisCircularBufferDAO.class)
+                      .add(key, valueAsString, maxSize);
+                if (!success) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public boolean removeAll(Collection<?> c) {
+        throw new UnsupportedOperationException("removeAll not supported by this circular buffer");
+    }
+
+    @Override
+    public boolean retainAll(Collection<?> c) {
+        throw new UnsupportedOperationException("retainAll not supported by this circular buffer");
+    }
+
+    @Override
+    public void clear() {
+        try (final Handle handle = rdbi.open()) {
+            handle.jedis().del(key);
+        }
+    }
+
+    @Override
+    public boolean offer(ValueType value) {
+        return addAll(ImmutableList.of(value));
+    }
+
+    @Override
+    public ValueType remove() {
+        ValueType element = poll();
+        if (element == null) {
+            throw new NoSuchElementException("Circular buffer is empty, cannot remove first element");
+        } else {
+            return element;
+        }
+    }
+
+    @Override
+    public ValueType poll() {
+        try (Handle handle = rdbi.open()) {
+            final String removedStr = handle.jedis().lpop(key);
+            if (removedStr == null) {
+                return null;
+            }
+            final ValueType removedValue = serializationHelper.decode(removedStr);
+            return removedValue;
+        }
+    }
+
+    @Override
+    public ValueType element() {
+        ValueType element = peek();
+        if (element == null) {
+            throw new NoSuchElementException("Circular buffer is empty, cannot retrieve head");
+        } else {
+            return element;
+        }
+    }
+
+    @Override
+    public ValueType peek() {
+        try (final Handle handle = rdbi.open()) {
+            final String removedStr = handle.jedis().lindex(key, 0);
+            if (removedStr == null) {
+                return null;
+            }
+            final ValueType removedValue = serializationHelper.decode(removedStr);
+            return removedValue;
+        }
+    }
+
+    public List<ValueType> peekAll() {
+        try (final Handle handle = rdbi.open()) {
+            final Pipeline pipeline = handle.jedis().pipelined();
+            List<ValueType> currentList = new ArrayList<ValueType>();
+            final int size = handle.jedis().llen(key).intValue();
+            final List<Response<String>> responses = Lists.newArrayListWithCapacity(size);
+
+            for (int i = 0; i < size; i++) {
+                responses.add(pipeline.lindex(key, i));
```

Comment:
i think it is probably better to just get the whole thing as lrange(...) in one shot, and probably best to not bother fetching the size
(again, doing it as a two-step you are leaving yourself open to concurrency problems. imagine that while thread A is in the middle of iterating through a peekAll, thread B starts rapidly popping members out.)